### ISSUE TITLE
how to interactively view？

### ISSUE THREAD

**Issue by LunaBright** [louhz/robogs#4]
Can you elaborate on how you can interactively view 4D renderings, change camera views, time frames, and control signals to see editing results?

**Comment by louhz** [comment]
The code of editing result for the interactive viewer  are pretty old now.

But i open the gui in the code base :    @torch.no_grad()
    def eval(self, step: int, stage: str = "val", ply_file: str = None, outpath: str = None):
        """Entry for evaluation."""
        print("Running evaluation...")
        cfg = self.cfg
        device = self.device

        if outpath is None:
            raise ValueError("outpath parameter cannot be None")

        if os.path.exists(outpath):
            warnings.warn(f"Output path '{outpath}' already exists. Files may be overwritten.")
        else:
            os.makedirs(outpath)

        trainloader = torch.utils.data.DataLoader(
            self.trainset, batch_size=1, shuffle=False, num_workers=1
        )

        camtoworlds1 = torch.tensor(
            [[[-0.6800, 0.4781, -0.5559, 1.8009],
            [0.1233, -0.6728, -0.7295, 1.4379],
            [-0.7228, -0.5646, 0.3985, -0.9330],
            [0.0000, 0.0000, 0.0000, 1.0000]]], device='cuda:0'
        )

        camtoworlds2 = torch.tensor(
            [[[0.3487, 0.7220, -0.5976, 1.8779],
            [0.5092, -0.6813, -0.5259, 1.4589],
            [-0.7869, -0.1209, -0.6052, 1.3671],
            [0.0000, 0.0000, 0.0000, 1.0000]]], device='cuda:0'
        )

        camtoworlds3 = torch.tensor(            [[[-0.0473,  0.6528, -0.7560,  1.9752],
            [ 0.1960, -0.7361, -0.6479,  1.1878],
            [-0.9795, -0.1788, -0.0932,  0.0921],
            [ 0.0000,  0.0000,  0.0000,  1.0000]]], device='cuda:0')

        data = trainloader.dataset[0]
        camtoworlds = camtoworlds3

        Ks = data["K"].to(device).view(1, 3, 3)
        pixels = data["image"].to(device) / 255.0
        pixels = pixels.view(1, pixels.shape[0], pixels.shape[1], 3)
        masks = data.get("mask", None)
        if masks is not None:
            masks = masks.to(device)

        height, width = pixels.shape[1:3]

        torch.cuda.synchronize()

        xyz, features_dc, features_extra, opacities, scales, rots, semantic_id = load_ply_sam(ply_file)

        stages = [
            # ("deform_arm", deform_arm),
            ("deform_hand_object", deform_scene_combined),
        ]
        idx=0
        for stage, deform_func in stages:
            for step, (xyz_out, opacities_out, scales_out, fextra_out, rots_out, fdc_out, sem_out) in enumerate(
                deform_func(xyz, features_dc, features_extra, opacities, scales, rots, semantic_id)
            ):
                self.render_step(
                    xyz_out, rots_out, scales_out, opacities_out, fdc_out,
                    fextra_out, device, camtoworlds, Ks, width, height, masks, cfg, stage, step,
                    frame=idx,
                    out_path=outpath
                )


as you can see here, the cam2worlds are the camera view you can edit on, and the deform_func such as deform_scene_combined are the function you can edit for motion changes .
